### ISSUE TITLE
Version 15.0.4 - Issue on writing objects to a streamWriter

### ISSUE THREAD

**Issue by luccasmf** [JoshClose/CsvHelper#1499]
I'm trying to read some CSV file from a URL then changing the delimiter and writing it file in my own folders, but everytime I try to write it, it brokes the record in a specific record.

When reading it, each record is well transformed for an dynamic list, but writing it again brokes the file. I have already tried to don't change the delimiter as well, and the same error happens.

```
public static byte[] CSVToTSV(byte[] csvFile, string csvName, out string TSVName)
        {
            byte[] tsvArray;
           
            using (var ms = new MemoryStream(csvFile))
            using (var reader = new StreamReader(ms, true))
            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
            using (var memoryStream = new MemoryStream())
            using (var streamWriter = new StreamWriter(memoryStream, Encoding.UTF8))
            using (var csvWriter = new CsvWriter(streamWriter, CultureInfo.InvariantCulture))
            {

                List<dynamic> records = new List<dynamic>();
                while (csv.Read())
                {
                    records.Add(csv.GetRecord<dynamic>());
                }


                csvWriter.Configuration.Delimiter = "\t";
                int i = 0;
                foreach (var record in records)
                {

                    csvWriter.WriteRecord(record);
                    csvWriter.NextRecord();
                    i++;
                }


                // csvWriter.WriteRecordsAsync(records);
                csvWriter.Flush();

                tsvArray = memoryStream.ToArray();


            }

            TSVName = csvName.Replace("csv", "tsv");

            return tsvArray;
        }
```

Ex: 

The output final lines are being showing like this:

> 2020-04-12	Brazil	MT	TOTAL	0	3	2	123	3	123
> 2020-04-12	Brazil	PA	TOTAL	3	13	42	259	13	246
> 2020-04-12	Brazil	PB	TOTAL	0	13	10	111	13	101
> 2020-04-12	Brazil	PE	TOTAL	13	85	144	960	85	960
> 2020-04-12	B

There was supposed to have more lines.

**Comment by JoshClose** [comment]
You need to flush the `StreamWriter`, not `CsvWriter`.

**Comment by JoshClose** [comment]
Also, I would suggest using the `CsvParser` directly for what you're doing. Here is an example that you can translate to using `MemoryStream`s.

```cs
void Main()
{
	var s = new StringBuilder();
	s.AppendLine("Id,Name");
	s.AppendLine("1,one");
	s.AppendLine("2,two");
	using (var reader = new StringReader(s.ToString()))
	using (var writer = new StringWriter())
	using (var csvParser = new CsvParser(reader, CultureInfo.InvariantCulture))
	using (var csvWriter = new CsvWriter(writer, CultureInfo.InvariantCulture))
	{
		csvWriter.Configuration.Delimiter = "\t";
		
		string[] record;
		while ((record = csvParser.Read()) != null)
		{
			foreach (var field in record)
			{
				csvWriter.WriteField(field);
			}

			csvWriter.NextRecord();
		}
		
		writer.ToString().Dump();
	}
}

public class Foo
{
	public int Id { get; set; }
	public string Name { get; set; }
}
```

**Comment by luccasmf** [comment]
Shame on me, I didn't even noticed I was Flushing the wrong writer. 

I need to use the writer because, after all, I just need the  byte array to send to Azure Datalake.

THanks for your kindness bro, it has worked by flushing the streamwriter.

I thought it was a bug of the version, since I have downgraded to 14.0 and it has worked, but I didn't notice that I have added the "streamWriter.Flush( )" in the same build.

Great job with this lib :D
